### PR TITLE
Fix/cmake files refactoring

### PR DESCRIPTION
**Pull request by istoilovgithub**
Refactor CMake files

Merge components refactoring from Windows port barnch
Move CMake modules and helpers under tools

Related tasks [APPLINK-30588](https://adc.luxoft.com/jira/browse/APPLINK-30588) [APPLINK-30972](https://adc.luxoft.com/jira/browse/APPLINK-30972)

### CODE REVIEW

**Comment by anosach-luxoft** [comment]
@istoilovgithub such huge commit messages are not allowed. Please make it fine.

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,6 @@ only:
   - \/feature\/.+
```
Comment (anosach-luxoft) [inline .travis.yml]: @istoilovgithub why have you decided to change this file?

File: CMakeLists.txt [hunk 2]
```diff
@@ -40,9 +44,9 @@ option(EXTENDED_MEDIA_MODE "Turn on and off extended Madia Manager features rela
 option(BUILD_SHARED_LIBS "Build all libraries as shared (if ON) or static (if OFF)" OFF)
 option(BUILD_BT_SUPPORT "Bluetooth support" ON)
 option(BUILD_USB_SUPPORT "libusb support" ON)
-option(BUILD_BACKTRACE_SUPPORT "backtrace support" ON)
-option(BUILD_TESTS "Possibility to build and run tests" OFF)
-option(TELEMETRY_MONITOR "Enable profiling time test util" ON)
+option(BUILD_BACKTRACE_SUPPORT "backtrace support" OFF)
+option(BUILD_TESTS "Possibility to build and run tests" ON)
+option(TELEMETRY_MONITOR "Enable profiling time test util" OFF)
```
Comment (anosach-luxoft) [inline CMakeLists.txt]: @istoilovgithub I think you should not change default build configuration

File: CMakeLists.txt [hunk 3]
```diff
@@ -231,14 +228,26 @@ endif(USE_DISTCC)
...
 if(ENABLE_GCOV)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
   add_definitions(-DGCOV_ENABLED)
+  message(STATUS "GCOV enabled")
 endif()
+
 set(CMAKE_CXX_FLAGS_RELEASE " -DNDEBUG -s -O2")
 set(CMAKE_CXX_FLAGS_DEBUG   " -DDEBUG -g3 -ggdb3")
 
+if(QT_PORT)
```
Comment (anosach-luxoft) [inline CMakeLists.txt]: @istoilovgithub do we need to support `QT_BUILD`?

File: .travis.yml [hunk 4]
```diff
@@ -56,6 +56,10 @@ deploy:
 notifications:
   email:
     - AKutsan@luxoft.com
+    - AByzhynar@luxoft.com
```
Comment (AByzhynar) [inline .travis.yml]: @istoilovgithub Remove all newly added persons from email notifications

File: .travis.yml [hunk 5]
```diff
@@ -20,10 +19,10 @@ addons:
     - libbluetooth3
     - libbluetooth-dev
     - libudev-dev
+    - libusb-1.0
     - cmake
     - html2text
     - clang-format-3.6
-    - binutils
```
Comment (AByzhynar) [inline .travis.yml]: @istoilovgithub Why did you delete it?

File: src/3rd_party-static/CMakeLists.txt [hunk 6]
```diff
@@ -32,8 +32,8 @@
 add_subdirectory(./jsoncpp)
 set(JSONCPP_INCLUDE_DIRECTORY ${JSONCPP_INCLUDE_DIRECTORY} PARENT_SCOPE)
 
-# --- MessageBroker
-add_subdirectory(./MessageBroker)
+# --- Message Broker
+add_subdirectory(./message_broker)
```
Comment (anosach-luxoft) [inline src/3rd_party-static/CMakeLists.txt]: @istoilovgithub did you change MessageBroker path?

File: src/components/application_manager/test/message_helper/CMakeLists.txt.user [hunk 7]
```diff
@@ -1,189 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (anosach-luxoft) [inline src/components/application_manager/test/message_helper/CMakeLists.txt.user]: @istoilovgithub what is it?

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc [hunk 8]
```diff
@@ -49,6 +49,10 @@
 #include "application_manager/mock_event_dispatcher.h"
 #include "application_manager/mock_state_controller.h"
 
+#if defined(_MSC_VER)
```
Comment (anosach-luxoft) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @istoilovgithub do we need to support microsoft compilator?

File: src/3rd_party-static/message_broker/CMakeLists.txt [hunk 9]
```diff
@@ -28,24 +28,51 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-function(create_cotired_test NAME SOURCES LIBS)
```
Comment (AByzhynar) [inline src/3rd_party-static/message_broker/CMakeLists.txt]: @istoilovgithub Why did you delete this function?

**Comment by anosach-luxoft** [comment]
@istoilovgithub I propose you to rework this PR. First of all, please remove from PR files not related to your task and temporary files. Also, please split changes to logical separated commits.

File: CMakeLists.txt (hunk 3 above)
Comment (istoilovgithub) [inline CMakeLists.txt]: @anosach-luxoft done by mistake, will remove it

File: src/3rd_party-static/CMakeLists.txt (hunk 6 above)
Comment (istoilovgithub) [inline src/3rd_party-static/CMakeLists.txt]: @anosach-luxoft this change has been made in pre_5_0_0_modifications, and looks like conversion all 3-rd part libraries to have names in lower case

File: src/components/application_manager/test/message_helper/CMakeLists.txt.user (hunk 7 above)
Comment (istoilovgithub) [inline src/components/application_manager/test/message_helper/CMakeLists.txt.user]: @anosach-luxoft done by mistake, will be removed

File: src/3rd_party-static/message_broker/CMakeLists.txt (hunk 9 above)
Comment (istoilovgithub) [inline src/3rd_party-static/message_broker/CMakeLists.txt]: @AByzhynar moved to tools/cmake/helpers/sources.cmake

File: .travis.yml (hunk 5 above)
Comment (istoilovgithub) [inline .travis.yml]: @AByzhynar basically the task is about to replay the refactoring that has been done in pre_5_0_0_modifications branch and this one might be wrong, will put it back.

**Comment by istoilovgithub** [comment]
@anosach-luxoft reworking the whole PR from scratch will take me at least an week, perhaps the commit message is not a brilliant one and I will replace it for sure. All current modifications are based on previous previous commits and are applied to the current branch by cherry-picks. Of course, some of the changes might be wrong, but let me know which ones and I will fix them too.

**Comment by istoilovgithub** [comment]
@anosach-luxoft fixed commit message, see https://github.com/smartdevicelink/sdl_core/pull/1144/commits/16a403b3cca43e4b211525f4b6c80a43466b5e97

File: src/3rd_party-static/message_broker/CMakeLists.txt (hunk 9 above)
Comment (istoilovgithub) [inline src/3rd_party-static/message_broker/CMakeLists.txt]: @AByzhynar moved to tools/cmake/helpers/sources.cmake as common module

File: CMakeLists.txt (hunk 3 above)
Comment (istoilovgithub) [inline CMakeLists.txt]: @anosach-luxoft fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/82c9705d98e5cd73c1907857ba208ca5092ea9c5

File: CMakeLists.txt (hunk 2 above)
Comment (istoilovgithub) [inline CMakeLists.txt]: @anosach-luxoft fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/82c9705d98e5cd73c1907857ba208ca5092ea9c5

File: .travis.yml (hunk 1 above)
Comment (istoilovgithub) [inline .travis.yml]: @anosach-luxoft fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/82c9705d98e5cd73c1907857ba208ca5092ea9c5

File: .travis.yml (hunk 4 above)
Comment (istoilovgithub) [inline .travis.yml]: @AByzhynar fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/82c9705d98e5cd73c1907857ba208ca5092ea9c5

File: src/components/application_manager/test/resumption/resume_ctrl_test.cc (hunk 8 above)
Comment (istoilovgithub) [inline src/components/application_manager/test/resumption/resume_ctrl_test.cc]: @anosach-luxoft fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/82c9705d98e5cd73c1907857ba208ca5092ea9c5

File: src/components/transport_manager/test/raw_message_matcher.cc [hunk 10]
```diff
@@ -30,7 +30,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "include/raw_message_matcher.h"
+#include "include/transport_manager/raw_message_matcher.h"
 //#include "../../include/protocol/raw_message.h"
```
Comment (AByzhynar) [inline src/components/transport_manager/test/raw_message_matcher.cc]: @istoilovgithub Commented code? What for?

File: src/components/utils/CMakeLists.txt [hunk 11]
```diff
@@ -28,62 +28,77 @@
...
+
 if(ENABLE_LOG)
-  list(APPEND SOURCES
-    ${UTILS_SRC_DIR}/push_log.cc
-    ${UTILS_SRC_DIR}/log_message_loop_thread.cc
-    ${UTILS_SRC_DIR}/logger_status.cc
-    ${UTILS_SRC_DIR}/auto_trace.cc
-    ${UTILS_SRC_DIR}/logger.cc
+  list(APPEND LIBRARIES
+	log4cxx -L${LOG4CXX_LIBS_DIRECTORY}
```
Comment (AByzhynar) [inline src/components/utils/CMakeLists.txt]: @istoilovgithub Indents?

File: src/components/media_manager/CMakeLists.txt [hunk 12]
```diff
@@ -99,22 +46,60 @@ include_directories (
...
+  list(APPEND LIBRARIES
+    ${GSTREAMER_gstreamer_LIBRARY}
+    pulse-simple
+    pulse
+    gobject-2.0
+    glib-2.0
+  )
+else()
+  list(APPEND EXCLUDE_PATHS
+	a2dp_source_player_adapter.cc
```
Comment (istoilovgithub) [inline src/components/media_manager/CMakeLists.txt]: @AByzhynar I am stick with 2 spaces for ident. AFAIK (as far as I know) unlike to c++ sources there is no strict coding standard allied to cmake files, isn't it?

File: src/components/resumption/test/CMakeLists.txt [hunk 13]
```diff
@@ -50,13 +52,4 @@ set(LIBRARIES
     ConfigProfile
```
Comment (istoilovgithub) [inline src/components/resumption/test/CMakeLists.txt]: @AByzhynar of cource do not, will be removed

File: src/components/transport_manager/test/raw_message_matcher.cc (hunk 10 above)
Comment (istoilovgithub) [inline src/components/transport_manager/test/raw_message_matcher.cc]: @AByzhynar came by automatic merge will be removed

File: src/components/media_manager/CMakeLists.txt (hunk 12 above)
Comment (AByzhynar) [inline src/components/media_manager/CMakeLists.txt]: @istoilovgithub Maybe Github shows it wrongly. But as I can see it looks like: 
```
list(APPEND LIBRARIES
    ${GSTREAMER_gstreamer_LIBRARY}
    pulse-simple
    pulse
    gobject-2.0
    glib-2.0
  )
else()
  list(APPEND EXCLUDE_PATHS
	a2dp_source_player_adapter.cc
	from_mic_recorder_adapter.cc
	from_mic_to_file_recorder_thread.cc
  )
endif()

```
So I can see 2 spaces here but 6 are present. I just prefer consistency. AFAIK for the moment we don't use strict rules for CMakeLists.txt . But as idea - we can try to implement usage and check style with something like: https://github.com/richq/cmake-lint

File: src/components/hmi_message_handler/test/CMakeLists.txt [hunk 14]
```diff
@@ -28,35 +28,40 @@
...
+set(EXCLUDE_PATHS)
+
 set(LIBRARIES    
     gmock
     ApplicationManager
     HMIMessageHandler
     jsoncpp
-    MessageBrokerClient
-    MessageBrokerServer
+	${BROKER_LIBRARIES}
```
Comment (istoilovgithub) [inline src/components/hmi_message_handler/test/CMakeLists.txt]: @AByzhynar BROKER_LIBRARIES just remains blank, so it will not cause any change to the library list

File: src/components/media_manager/CMakeLists.txt (hunk 12 above)
Comment (istoilovgithub) [inline src/components/media_manager/CMakeLists.txt]: @AByzhynar I am completely agree with you that having some consistency in style is essential and applying style checker make perfectly sense. Until then enforcing the consistency is just a personal responsibility for each contributor.

File: src/components/connection_handler/CMakeLists.txt [hunk 15]
```diff
@@ -39,20 +38,20 @@ include_directories (
     ${LOG4CXX_INCLUDE_DIRECTORY}
 )
 
-set (SOURCES
-    ${CH_SRC_DIR}/connection_handler_impl.cc
-    ${CH_SRC_DIR}/connection.cc
-    ${CH_SRC_DIR}/device.cc
-    ${CH_SRC_DIR}/heartbeat_monitor.cc
+set(PATHS
+  ${CMAKE_CURRENT_SOURCE_DIR}/include
```
Comment (LuxoftAKutsan) [inline src/components/connection_handler/CMakeLists.txt]: @istoilovgithub are you sure that sources are in include dirrectory?

File: tools/cmake/modules/FindSDLSqlite3.cmake [hunk 16]
```diff
@@ -0,0 +1,15 @@
+#  - Try to find Sqlite3
+#
+#  SQLITE3_INCLUDE_DIRECTORY - the Sqlite3 include directory
+#  SQLITE3_LIBRARIES - the Sqlite3 libraries
+
+get_os(OS)
+if(${OS} STREQUAL "posix")
+  find_package(Sqlite3 REQUIRED)
+elseif(${OS} STREQUAL "win")
```
Comment (LuxoftAKutsan) [inline tools/cmake/modules/FindSDLSqlite3.cmake]: @istoilovgithub please remove win and qt dependencies. Qt and Win only in pre_5_0

File: src/components/connection_handler/CMakeLists.txt (hunk 15 above)
Comment (istoilovgithub) [inline src/components/connection_handler/CMakeLists.txt]: @LuxoftAKutsan As far as I understood that is one of the main objective for this refactoring - simplifying the cmake files by using common functions. So my answer there is no guarantee since there is no explicit reference to each source files, but the correct content should be ensured implicitly (by definition), right?

File: tools/cmake/modules/FindSDLSqlite3.cmake (hunk 16 above)
Comment (istoilovgithub) [inline tools/cmake/modules/FindSDLSqlite3.cmake]: @LuxoftAKutsan Sure, will drop all related

File: src/components/connection_handler/CMakeLists.txt (hunk 15 above)
Comment (istoilovgithub) [inline src/components/connection_handler/CMakeLists.txt]: @LuxoftAKutsan fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/22b926d4af4e4aab9ae0746d2bfd8625ceadcc9d

File: src/components/resumption/test/CMakeLists.txt (hunk 13 above)
Comment (istoilovgithub) [inline src/components/resumption/test/CMakeLists.txt]: @AByzhynar fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/22b926d4af4e4aab9ae0746d2bfd8625ceadcc9d

File: src/components/transport_manager/test/raw_message_matcher.cc (hunk 10 above)
Comment (istoilovgithub) [inline src/components/transport_manager/test/raw_message_matcher.cc]: @AByzhynar fixed with https://github.com/smartdevicelink/sdl_core/pull/1144/commits/22b926d4af4e4aab9ae0746d2bfd8625ceadcc9d
